raw-svo.empty.lua

```diff
@@ -97,17 +97,40 @@ empty.focus = function()
-empty.treecurableswithmadness = {"ablaze", "aeon", "agoraphobia", "anorexia", "asthma", "blackout", "claustrophobia", "clumsiness", "crippledleftarm", "crippledleftleg", "crippledrightarm", "crippledrightleg", "darkshade", "deadening", "disloyalty", "disrupt", "dissonance", "dizziness", "epilepsy", "fear", "generosity", "haemophilia", "healthleech",  "hellsight", "hypochondria", "inlove", "itching", "justice", "pacifism", "paralysis", "peace", "relapsing", "selarnia", "sensitivity", "shyness", "slickness", "stuttering", "unknownany", "unknowncrippledarm", "unknowncrippledleg", "unknownmental", "vertigo", "voyria", "weakness", "hamstring", "shivering", "frozen", "depression", "parasite", "retribution", "shadowmadness", "timeloop", "degenerate", "deteriorate"}
-
+empty.treecurables = {"ablaze", "addiction", "aeon", "agoraphobia", "anorexia", "asthma", "blackout", "claustrophobia", "clumsiness", "confusion", "crippledleftarm", "crippledleftleg", "crippledrightarm", "crippledrightleg", "darkshade", "deadening", "dementia", "disloyalty", "disrupt", "dissonance", "dizziness", "epilepsy", "fear", "generosity", "haemophilia", "hallucinations", "healthleech",  "hellsight", "hypersomnia", "hypochondria", "illness", "impatience", "inlove", "itching", "justice", "lethargy", "loneliness", "madness", "masochism", "pacifism", "paralysis", "paranoia", "peace", "recklessness", "relapsing", "selarnia", "sensitivity", "shyness", "slickness", "stupidity", "stuttering", "unknownany", "unknowncrippledarm", "unknowncrippledleg", "unknownmental", "vertigo", "voyria", "weakness", "hamstring", "shivering", "frozen", "skullfractures", "crackedribs", "wristfractures", "torntendons", "depression", "parasite", "retribution", "shadowmadness", "timeloop", "degenerate", "deteriorate"}
+empty.treeblocks = {
+  madness = {"madness", "dementia", "stupidity", "confusion", "hypersomnia", "paranoia", "hallucinations", "impatience", "addiction", "agoraphobia", "inlove", "loneliness", "recklessness", "masochism"},
+  hypothermia = {"frozen", "shivering"},
+}
 -- expose publicly
 treecurables = empty.treecurables
-empty.tree = function ()
-  if affs.madness then
-    removeaff(empty.treecurableswithmadness)
-  else
-    removeaff(empty.treecurables)
+
+gettreeableaffs = function(getall)
+  local a = deepcopy(empty.treecurables)
+  for blockaff, blocked in pairs(empty.treeblocks) do
+    if affs[blockaff] then
+      for index, remaff in ipairs(blocked) do
+        table.remove(a, table.index_of(a, remaff))
+      end
+    end
   end
+  if not getall then
+    local i = 1
+    while #a >= i do
+      if not affs[a[i]] then
+        table.remove(a, i)
+      else
+        i = i + 1
+      end
+    end
+  end
+  return a
+end
+
+empty.tree = function ()
+  local a = gettreeableaffs(true)
```

Comment:
`if` not getall then
` -- stuff to remove afflictions we don't have

True returns all afflictions that are treeable. No input returns treeable afflictions that we have.